differ/differ_test.go

```diff
@@ -411,21 +410,22 @@ func TestSetupNotificationProducerValidBrokerConf(t *testing.T) {
 	assert.Nil(t, err, "Unexpected behavior: Producer was not closed successfully")
 }
 
-func TestSetupNotificationProducerDisabledBrokerConfig(t *testing.T) {
-	testConfig := conf.ConfigStruct{
-		Kafka: conf.KafkaConfiguration{
-			Enabled: false,
-		},
-	}
-
-	setupKafkaProducer(testConfig)
-
-	msgBytes, err := json.Marshal(types.NotificationMessage{})
-	helpers.FailOnError(t, err)
+func TestStartDifferNoDestination(t *testing.T) {
+	if os.Getenv("SETUP_PRODUCER") == "1" {
```

Comment:
why env. variables?

Reply:
this is a "trick" to be able to catch the exit code within a UT. When the test is run, it skips lines 415 and 416, does the whole `exec.Command` thing, which calls the test again and we can capture the exit code of the command.